ContextDesktop/src/main/java/com/github/migi_1/Context/model/MainEnvironment.java

```diff
@@ -0,0 +1,249 @@
+        fpp.addFilter(dlsf);
+        viewPort.addProcessor(fpp);
+    }
+
+    /**
+     * Initializes all objects and translations/rotations of the scene.
+     */
+    private void initSpatials() {
+
+        levelGenerator = new LevelGenerator(WORLD_LOCATION);
+        platform = new Platform(PLATFORM_LOCATION);
+        commander = new Commander(COMMANDER_LOCATION);
+
+        //attach all objects to the root pane
+        for (LevelPiece levelPiece : levelGenerator.getLevelPieces(COMMANDER_LOCATION)) {
+            addDisplayable(levelPiece);
+        }
+
+        addEntity(platform);
+        addEntity(commander);
+    }
+
+    /**
+     * Initializes the cameras and sets its location and rotation.
+     * Starts with the VR camera.
+     */
+    private void initCameras() {
+        commander.getModel().rotate(0f, COMMANDER_ROTATION, 0f);
+        flyObs.getModel().setLocalTranslation(new Vector3f(-12f, 0f, -16f));
+        flyObs.getModel().setLocalRotation(new Quaternion(0f, 0f, 0f, 1f));
+
+        commander.makeObserver();
+        addEntity(flyObs);
+    }
+
+    /**
+     * Renders the entities.
+     * @param rm manager of the renderengine
+     */
+    @Override
+    public void render(RenderManager rm) { }
+
+    /**
+     * Moves the flycam.
+     * @param move a vector representation of the movement of the flyCamera.
+     */
+    public void moveCam(Vector3f move) {
+        flyObs.getModel().move(move);
+    }
+
+    /**
+     * Rotates the flycam.
+     * @param rotation the rotation vector.
+     */
+    public void rotateCam(Vector3f rotation) {
+        flyObs.getModel().rotate(rotation.x, rotation.y, rotation.z);
+    }
+
+    /**
+     * Returns the current camera (which is a observer).
+     * @return A string representation of the current camera:
+     * "VR (Node)" or "FLY (Node)".
+     */
+    public boolean getFlyCamActive() {
+        return flyCamActive;
+    }
+
+    /**
+     * Swap the cameras in the environment.
+     * VRCam <-> FlyCam.
+     */
+    public void swapCamera() {
+        if (!flyCamActive) {
+            VRApplication.setObserver(flyObs.getModel());
+            flyCamActive = true;
+        } else {
+            VRApplication.setObserver(commander.getModel());
+            flyCamActive = false;
+        }
+    }
+
+    /**
+     * Updates the test world.
+     */
+    private void updateTestWorld() {
+
+        //delete level piece when it too far back
+        for (LevelPiece levelPiece : levelGenerator.deleteLevelPieces(commander.getModel().getLocalTranslation())) {
+        	removeDisplayable(levelPiece);
+        }
+        for (LevelPiece levelPiece : levelGenerator.getLevelPieces(commander.getModel().getLocalTranslation())) {
+        	addDisplayable(levelPiece);
+        }
+    }
+
+    /**
+     * Update the steering direction.
+     * @param orientation This translates to the steering ange.
+     */
+    public void steer(float orientation) {
+        steering = orientation;
```

Comment:
Maybe this is not the right place for this method. Steering doesn't influence the visual aspects of the world.